api/app/controllers/api/v1/passwords_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Api
+  module V1
+    # Passwords controller
+    class PasswordsController < ApplicationController
+      include Password
+
+      def new
+        user = User.find_by(email: params[:email])
+        return user_not_found_error unless user
+        user.process_reset_password_request
+      end
+
+      def update
+        user = User.by_reset_token(params[:reset_token]).first
+        return user_not_found_error unless user
+        return invalid_token_error unless user.valid_token?
+        user.update_password(params[:password], params[:password_confirmation])
+        if user.save
+          render status: :no_content
+        else
+          render json: { errors: user.errors.full_messages },
```

Comment:
This seems wrong. Error responses need to follow the correct format for JSON API. Look at the render_single_resource method in the resourceful concern. If there is an error on a resource, we return it with the error serializer, that gives the error a specific format. You're following a different convention here, which feels hacky.